apps/api/src/recurring-donation/recurring-donation.service.ts

```diff
@@ -39,15 +144,60 @@ export class RecurringDonationService {
         extCustomerId: updateRecurringDonationDto.extCustomerId,
         amount: updateRecurringDonationDto.amount,
         currency: updateRecurringDonationDto.currency,
+        vaultId: updateRecurringDonationDto.vaultId,
       },
     })
     if (!result) throw new NotFoundException('Not found')
     return result
   }
 
+  async cancel(id: string): Promise<RecurringDonation | null> {
+    return await this.updateStatus(id, RecurringDonationStatus.canceled)
+  }
+
   async remove(id: string): Promise<RecurringDonation | null> {
     const result = await this.prisma.recurringDonation.delete({ where: { id: id } })
     if (!result) throw new NotFoundException('Not found')
     return result
   }
+
+  async findSubscriptionByExtId(extSubscriptionId: string): Promise<RecurringDonation | null> {
+    const result = await this.prisma.recurringDonation.findFirst({
+      where: { extSubscriptionId },
+      select: RecurringDonationService.defaultSelectFields,
+    })
+    return result
+  }
+
+  async updateStatus(
+    id: string,
+    status: RecurringDonationStatus,
+  ): Promise<RecurringDonation | null> {
+    const result = await this.prisma.recurringDonation.update({
+      where: { id: id },
+      data: {
+        status: status,
+      },
+    })
+    if (!result) throw new NotFoundException('Not found')
```

Comment:
@slavcho This method should throw `BadRequestException` and not `NotFoundException` as it will send status code 404 to the client. 

We actually have a proper HTTP handler for the method, but the params that are sent to it are bad so we better respond with status code 400.

Reply:
Fixed. Thanks.